### ISSUE TITLE
--advertise-host does not work on CentOS7

### ISSUE THREAD

**Issue by qiukeren** [cockroachdb/cockroach#14694]
**BUG REPORT**

the config is

```
I170407 02:02:24.990694 1 util/log/clog.go:991  [config] file created at: 2017/04/07 02:02:24
I170407 02:02:24.990694 1 util/log/clog.go:991  [config] running on machine: HOSTNAME
I170407 02:02:24.990694 1 util/log/clog.go:991  [config] binary: CockroachDB CCL beta-20170406 (linux amd64, built 2017/04/06 15:55:10, go1.8)
I170407 02:02:24.990694 1 util/log/clog.go:991  [config] arguments: [cockroach start --store=node1 --port=26257 --host=0.0.0.0 --insecure --advertise-host=XXX.XXXX.XXX]
```

But: in node status I should get XXX.XXXX.XXX in address column, but i just get HOSTNAME instead.

```
$ cockroach node status
+----+----------------------+---------------+---------------------+---------------------+------------+-----------+-------------+--------------+--------------+------------------+-----------------------+--------+--------------------+------------------------+
| id |       address        |     build     |     updated_at      |     started_at      | live_bytes | key_bytes | value_bytes | intent_bytes | system_bytes | replicas_leaders | replicas_leaseholders | ranges | ranges_unavailable | ranges_underreplicated |
+----+----------------------+---------------+---------------------+---------------------+------------+-----------+-------------+--------------+--------------+------------------+-----------------------+--------+--------------------+------------------------+
|  1 | HOSTNAME:26257 | beta-20170406 | 2017-04-07 02:00:57 | 2017-04-07 02:00:57 |      20866 |       863 |       20556 |          483 |         1123 |                0 |                     0 |      0 |                  0 |                      0 |
+----+----------------------+---------------+---------------------+---------------------+------------+-----------+-------------+--------------+--------------+------------------+-----------------------+--------+--------------------+------------------------+
(1 row)

```

MacOS sierra does not have this issue.

2. Please describe the issue you observed:

This issue happen on two different machines (both Centos7) (2 machines in total).

This issue happen on both the master branch or official latest binary build.

**Comment by mberhault** [comment]
Apparently we broke it. There's a PR out to fix it, but it's not merged yet: #14692 
You can try a build after checkout that out PR, or wait until it's merged, but it should fix the issue.

**Comment by a-robinson** [comment]
Yup, I'm sorry @qiukeren! I'll have the fix checked in soon.
